src/Symfony/Component/HttpKernel/DataCollector/ConfigDataCollector.php

```diff
@@ -270,60 +270,20 @@ public function getName()
      *
      * @return string One of: unknown, dev, stable, eom, eol
      */
-    private function requestSymfonyState()
+    private function calculateSymfonyState()
     {
-        $versionInfo = null;
-
-        // get version information from cache or the roadmap
-        $versionCachePath = $this->kernel->getCacheDir().'/version_info.json';
-        if (file_exists($versionCachePath)) {
-            $versionInfo = json_decode(file_get_contents($versionCachePath), true);
+        $now = new \DateTime();
+        $eom = \DateTime::createFromFormat('m/Y', Kernel::END_OF_MAINTENANCE)->modify('last day of this month');
+        $eol = \DateTime::createFromFormat('m/Y', Kernel::END_OF_LIFE)->modify('last day of this month');
+
+        if ($now > $eol) {
+            $versionState = 'eol';
+        } elseif ($now > $eom) {
+            $versionState = 'eom';
+        } elseif ('DEV' === Kernel::EXTRA_VERSION) {
```

Comment:
well I know I wrote this condition, I think we should change it to `elseif ('' !== Kernel::EXTRA_VERSION) {` and change the name from `dev` to `unstable`. This is because currently, you get "stable" as state when using a BETA or RC release, which I think is wrong.